File: app/src/main/java/com/liushu/crazyandroid/ui/stage01/chapter02/activity/Demo020306Activity.java
package com.liushu.crazyandroid.ui.stage01.chapter02.activity;

import android.widget.CompoundButton;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.Switch;
import android.widget.TextView;
import android.widget.ToggleButton;

import com.jaydenxiao.common.base.BaseActivity;
import com.liushu.crazyandroid.R;

import butterknife.BindView;
import butterknife.OnClick;

public class Demo020306Activity extends BaseActivity {
    @BindView(R.id.iv_back)
    ImageView mIvBack;
    @BindView(R.id.tv_title_name)
    TextView mTvTitleName;
    @BindView(R.id.tb_test)
    ToggleButton mTbTest;
    @BindView(R.id.switch_test)
    Switch mSwitchTest;
    @BindView(R.id.ll_test)
    LinearLayout mLlTest;
    @BindView(R.id.activity_demo020306)
    LinearLayout mActivityDemo020306;

    @Override
    public int getLayoutId() {
        return R.layout.activity_demo020306;
    }

    @Override
    public void initView() {
        mTvTitleName.setText("动态控制布局");
        CompoundButton.OnCheckedChangeListener listener = new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                if (isChecked) {
                    mSwitchTest.setChecked(true);
                    mTbTest.setChecked(true);
                    mLlTest.setOrientation(LinearLayout.VERTICAL);
                } else {
                    mSwitchTest.setChecked(false);
                    mTbTest.setChecked(false);
                    mLlTest.setOrientation(LinearLayout.HORIZONTAL);
                }
            }
        };
        mSwitchTest.setOnCheckedChangeListener(listener);
        mTbTest.setOnCheckedChangeListener(listener);
    }

    @OnClick(R.id.iv_back)
    public void onClick() {
        finish();
    }
}

File: app/src/main/java/com/liushu/crazyandroid/ui/stage01/chapter02/activity/Demo020306Activity.kt
package com.liushu.crazyandroid.ui.stage01.chapter02.activity

import android.widget.CompoundButton
import android.widget.LinearLayout
import butterknife.OnClick
import com.jaydenxiao.common.base.BaseActivity
import com.liushu.crazyandroid.R
import kotlinx.android.synthetic.main.activity_demo020306.*
import kotlinx.android.synthetic.main.head_title_layout.*

class Demo020306Activity : BaseActivity() {

    override val layoutId: Int
        get() = R.layout.activity_demo020306

    override fun initView() {
        tv_title_name!!.text = "动态控制布局"
        val listener = CompoundButton.OnCheckedChangeListener { buttonView, isChecked ->
            if (isChecked) {
                switch_test!!.isChecked = true
                tb_test!!.isChecked = true
                ll_test!!.orientation = LinearLayout.VERTICAL
            } else {
                switch_test!!.isChecked = false
                tb_test!!.isChecked = false
                ll_test!!.orientation = LinearLayout.HORIZONTAL
            }
        }
        switch_test!!.setOnCheckedChangeListener(listener)
        tb_test!!.setOnCheckedChangeListener(listener)
    }

    @OnClick(R.id.iv_back)
    fun onClick() {
        finish()
    }
}
